core/base/src/TString.cxx

```diff
@@ -126,7 +126,8 @@ TString::TString(const std::string &s)
 TString::TString(const char *cs, Ssiz_t n)
 {
    if (n < 0) {
-      Error("TString::TString", "Negative length!");
+      char *data = Init(0, 0);
```

Comment:
I think the `Zero()` member method is more concise to initialize the empty string.